### ISSUE TITLE
[Bug] a11y-tooltip-interactive-trigger raises false positives on components

### ISSUE THREAD

**Issue by khiga8** [primer/eslint-plugin-primer-react#315]
When I tested turning this rule on in dotcom (https://github.com/github/github/pull/361996), I noticed that false positives are raised for custom components like:

```
<Tooltip>
   <HistoryLinkButton ..>
</Tooltip>
```

Currently the rule only allows certain components, and rejects whatever it doesn't recognize. 
A potential approach is to support an "allowed config" and have consumers add to it. I worry this might still be noisy though to consumers.

An alternative is to rewrite the rule to be more conservative and consider a denylist of components.
